extensions/amp-analytics/0.1/linker-manager.js

```diff
@@ -247,13 +268,62 @@ export class LinkerManager {
           Services.documentInfoForDoc(this.ampdoc_);
       const sourceOrigin = this.urlService_.parse(sourceUrl).hostname;
       const canonicalOrigin = this.urlService_.parse(canonicalUrl).hostname;
-      if (!areFriendlyDomains(sourceOrigin, hostname)
-          && !areFriendlyDomains(canonicalOrigin, hostname)) {
-        return;
+      if (!areFriendlyDomains(sourceOrigin, url)
+          && !areFriendlyDomains(canonicalOrigin, url)) {
+        return false;
       }
     }
+    return true;
+  }
+
+  /**
+   * Register callback that will handle form sumbits.
+   */
+  enableFormSupport_() {
```

Comment:
we will need to remove the listener when amp-analytics unlayout. (can be a separate PR).

Reply:
SGTM will make follow-up PR